src/interfaces/IOptionSettlementEngine.sol

```diff
@@ -49,6 +33,43 @@ interface IOptionSettlementEngine {
         uint40 indexed expiryTimestamp
     );
 
+    /**
+     * @notice Emitted when new options contracts are written.
+     * @param optionId The token id of the option type written.
+     * @param writer The address of the writer.
+     * @param claimId The claim token id of the new or existing short position written against.
+     * @param amount The amount of options contracts written.
+     */
+    event OptionsWritten(uint256 indexed optionId, address indexed writer, uint256 indexed claimId, uint112 amount);
+
+    /**
+     * @notice Emitted when options contracts are written into a bucket.
+     * @param optionId The token id of the option type written.
+     * @param bucketIndex The index of the bucket to which the options were written.
+     * @param amount The amount of options contracts written.
+     */
+    event BucketWrittenInto(uint256 indexed optionId, uint96 bucketIndex, uint112 amount);
```

Comment:
I think this should have the claim ID indexed for the subgraph handler to know.

Reply:
RIght on, good call, will add